app/containers/Admin/ApplicationReview/ReviewSidebar.tsx

```diff
@@ -0,0 +1,343 @@
+    const variables = {
+      input: {
+        id,
+        reviewCommentPatch: {
+          resolved: resolve
+        }
+      }
+    };
+    await updateReviewCommentMutation(environment, variables);
+  };
+  const deleteComment = async (id, commentType) => {
+    const {environment} = relay;
+    const variables = {
+      input: {
+        id,
+        reviewCommentPatch: {
+          deletedAt: nowMoment().format('YYYY-MM-DDTHH:mm:ss')
+        }
+      }
+    };
+
+    await deleteReviewCommentMutation(
+      environment,
+      variables,
+      applicationReviewStep.id,
+      `ReviewSidebar_${commentType}Comments`
+    );
+  };
+  return (
+    <div id="sidebar" className="col-md-5 col-lg-4 col-xxl-3">
+      <button
+        type="button"
+        id="close"
+        aria-label={`Close ${reviewStepName} Review`}
+        onClick={onClose}
+      >
+        ×
+      </button>
+      <h2>{reviewStepName} Review</h2>
+      {applicationReviewStep.isComplete ? (
+        <p className="mark-incomplete">
+          <FontAwesomeIcon icon={faCheck} style={{marginRight: 2}} />
+          <span> This review step has been completed. </span>
+          <Button
+            variant="link"
+            style={{
+              padding: '0 0 2px 0',
+              fontSize: '0.9rem',
+              lineHeight: 1,
+              display: 'inline'
+            }}
+            onClick={() => markReviewStepComplete(false)}
+          >
+            Mark incomplete
+          </Button>
+        </p>
+      ) : (
+        markCompletedButton
+      )}
+      <div id="scrollable-comments" tabIndex={0}>
+        <h3 id="general-comments-label">
+          General Comments{' '}
+          <small>
+            <em>(Visible to applicant)</em>
+          </small>
+        </h3>
+        {noGeneralCommentsToShow ? (
+          <p className="empty-comments">No general comments to show.</p>
+        ) : (
+          <ul aria-labelledby="general-comments-label">
+            {generalComments.map(
+              ({
+                node: {
+                  id,
+                  description,
+                  createdAt,
+                  ciipUserByCreatedBy,
+                  resolved
+                }
+              }) => {
+                const showComment = showingResolved || !resolved;
+                return !showComment ? null : (
+                  <ReviewComment
+                    key={id}
+                    id={id}
+                    description={description}
+                    createdAt={createdAt}
+                    createdBy={`${ciipUserByCreatedBy.firstName} ${ciipUserByCreatedBy.lastName}`}
+                    viewOnly={applicationReviewStep.isComplete}
+                    isResolved={resolved}
+                    onResolveToggle={resolveComment}
+                    onDelete={(id) => deleteComment(id, 'general')}
+                  />
+                );
+              }
+            )}
+          </ul>
+        )}
```

Comment:
Sonarcloud is complaining that this block is duplicated by the one below. It's a bit tricky because of the comment type in the `deleteComment` call. If we can't remove that parameter from `deleteComment`, we might need to write something like
```tsx
   const makeRenderReviewComment(commentType) => ({
                node: {
                  id,
                  description,
                  createdAt,
                  ciipUserByCreatedBy,
                  resolved
                }
              }) => {
                const showComment = showingResolved || !resolved;
                return !showComment ? null : (
                  <ReviewComment
                    key={id}
                    id={id}
                    description={description}
                    createdAt={createdAt}
                    createdBy={`${ciipUserByCreatedBy.firstName} ${ciipUserByCreatedBy.lastName}`}
                    viewOnly={applicationReviewStep.isComplete}
                    isResolved={resolved}
                    onResolveToggle={resolveComment}
                    onDelete={(id) => deleteComment(id, 'general')}
                  />
                );
              }
```
and use it like
```tsx
generalComments.map(makeRenderReviewComment('general'))
internalComments.map(makeRenderReviewComment('internal'))

```